contracts/utils/SignatureValidator.sol

```diff
@@ -1,195 +1,91 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.7.6;
 
-import "../interfaces/IERC1271Wallet.sol";
-import "./LibBytes.sol";
-
-interface IWallet {
-    /// @dev Verifies that a signature is valid.
-    /// @param hash Message hash that is signed.
-    /// @param signature Proof of signing.
-    /// @return isValid Validity of order signature.
-    function isValidSignature(bytes32 hash, bytes memory signature) external view returns (bool isValid);
+import { IERC1271Wallet } from "../interfaces/IERC1271Wallet.sol";
+import { LibBytes } from "./LibBytes.sol";
+import { ECDSA } from "@openzeppelin/contracts/cryptography/ECDSA.sol";
+
+// 0x1626ba7e
+bytes4 constant ERC1271_MAGICVALUE = bytes4(keccak256("isValidSignature(bytes32,bytes)"));
+
+// 0xb0671381
+bytes4 constant ZX1271_MAGICVALUE = bytes4(keccak256("isValidWalletSignature(bytes32,address,bytes)"));
+
+enum SignatureType {
+    Illegal, // 0x00, default as illegal
+    Invalid, // 0x01
+    EIP712, // 0x02 standard EIP-712 signature
+    EthSign, // 0x03 signed using web3.eth_sign() or Ethers wallet.signMessage()
+    WalletBytes, // 0x04 DEPRECATED, never used
+    Standard1271, // 0x05 standard EIP-1271 wallet type
```

Comment:
nit: I would prefer calling it `EIP1271`

Reply:
Fixed in 2366b87